From 7c9a315578e464d9be09f6cebf1d9d073731cd86 Mon Sep 17 00:00:00 2001
From: Sanchayan Maity <sanchayan@asymptotic.io>
Date: Thu, 24 Feb 2022 20:26:46 +0530
Subject: [PATCH] ldac: Set eqmid in caps

We set the eqmid in caps to be usable downstream by rtpldacpay for
knowing the frame count.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/1797>
---
 subprojects/gst-plugins-bad/ext/ldac/gstldacenc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ext/ldac/gstldacenc.c b/ext/ldac/gstldacenc.c
index 9ca2ef8fbf..cd82ae1396 100644
--- a/ext/ldac/gstldacenc.c
+++ b/ext/ldac/gstldacenc.c
@@ -231,6 +231,9 @@ gst_ldac_enc_do_negotiate (GstAudioEncoder * audio_enc)
   output_caps = gst_caps_fixate (output_caps);
   gst_caps_set_simple (output_caps, "framed", G_TYPE_BOOLEAN, TRUE, NULL);
 
+  /* Set EQMID in caps to be used downstream by rtpldacpay */
+  gst_caps_set_simple (output_caps, "eqmid", G_TYPE_INT, enc->eqmid, NULL);
+
   GST_INFO_OBJECT (enc, "output caps %" GST_PTR_FORMAT, output_caps);
 
   if (enc->channels == 1)
-- 
GitLab

